src/AgreementManager.ts

```diff
@@ -503,27 +482,14 @@ class AgreementManager {
             this.forks[forkCnt].agreements.length - 1
         ];
     }
-    private getAgreement(
-        forkCnt: number,
-        transactionCnt: number
-    ): Agreement | undefined {
-        if (Number(forkCnt) >= this.forks.length) return undefined;
-        let fork = this.forks[forkCnt];
-        if (transactionCnt >= fork.agreements.length) return undefined;
-        return fork.agreements[transactionCnt];
-    }
     //both canonical chain and future queue
     public isBlockInChain(block: BlockStruct): boolean {
-        let agreement = this.getAgreement(
-            Number(block.transaction.header.forkCnt),
-            Number(block.transaction.header.transactionCnt)
-        );
-        if (
-            agreement &&
-            EvmUtils.encodeBlock(agreement.block) == EvmUtils.encodeBlock(block)
-        )
-            return true;
-        return false;
+        const agreement = this.getAgreementByBlock(block);
+
+        return agreement
+            ? EvmUtils.encodeBlock(agreement.block) ==
+                  EvmUtils.encodeBlock(block)
+            : false;
```

Comment:
return agreement && this.areBlocksEqual(agreement.block, block)

Reply:
your suggestion gives this error:
`Type 'boolean | undefined' is not assignable to type 'boolean'.
  Type 'undefined' is not assignable to type 'boolean'.ts(2322)`
  
  how about 
  ```ts
      public isBlockInChain(block: BlockStruct): boolean {
        const agreement = this.getAgreementByBlock(block);
        return (
            (agreement || false) && this.areBlocksEqual(agreement.block, block)
        );
    }
    ```